### PR TITLE
[decomp] Use var_mean in native_batch_norm decomposition

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #94073
* #94522
* __->__ #94140
* #94139
* #94142
* #94072

### CODE REVIEW

**Comment by ngimel** [comment]
Did you check that inductor perf didn't change?

File: test/test_decomp.py [hunk 1]
```diff
@@ -167,7 +167,10 @@ def op_assert_ref(test_case, op, test_dtype, i, orig, decomp, ref, args, kwargs)
         (torch.float16, torch.ops.aten._native_batch_norm_legit.no_stats): 1e-5,
         (torch.bfloat16, torch.ops.aten.linalg_vector_norm.default): 1e-4,
         (torch.float16, torch.ops.aten.linalg_vector_norm.default): 1e-4,
+        (torch.bfloat16, torch.ops.aten.var_mean.correction): 5e-7,
```
Comment (ngimel) [inline test/test_decomp.py:170]: why does tolerance change here?

**Review by ngimel — APPROVED** [review]

**Comment by peterbell10** [comment]
> Did you check that inductor perf didn't change?

This improves perf by removing the duplicate mean calculation. However it's very slight since the second mean was being fused with the sum of square deviations in the variance. In the following example, I see a 0.6% speedup from 366 us to 364 us
```python
import torch
import torch._dynamo
from torch._inductor import config
config.debug = True

a = torch.nn.BatchNorm3d(10).train().cuda()
b = torch.rand(10, 10, 16, 64, 64, device="cuda")

@torch._dynamo.optimize()
def fn(x):
    return a(x)

_ = fn(b)
%timeit fn(b)

```

File: test/test_decomp.py (hunk 1 above)
Comment (peterbell10) [reply test/test_decomp.py:170]: `aten.var_mean` uses a different algorithm from `aten.mean` which ends up being slightly more precise.  `5e-7` is still incredibly good for half-precision though. The default `rtol` for `torch.testing` is `1e-5`.

**Pull request closed without merging**